### ISSUE TITLE
Android TV Detox test build fails for 3rd party libs

### ISSUE THREAD

**Issue by tikhonNikita** [react-native-tvos/react-native-tvos#504]
### New Version

0.66.3

### Old Version

0.71.8

### Build Target(s)

Android test build

### Output of `react-native info`

System:
    OS: macOS 13.0
    CPU: (12) arm64 Apple M2 Pro
    Memory: 72.55 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.9.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.4 - /opt/homebrew/bin/npm
    Watchman: 2023.04.10.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.0 - /Users/tikhonov/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    Android SDK:
      API Levels: 28, 31, 32, 33, 33
      Build Tools: 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.2
      System Images: android-28 | Google ARM64-V8a Play ARM 64 v8a, android-30 | Google TV Intel x86 Atom, android-30 | Google APIs ARM 64 v8a, android-31 | Android TV ARM 64 v8a, android-33 | Android TV ARM 64 v8a, android-33 | Android TV Intel x86 Atom, android-33 | Google TV ARM 64 v8a, android-33 | Google TV Intel x86 Atom, android-33 | Google APIs ARM 64 v8a, android-33 | Google APIs Intel x86_64 Atom, android-33 | Google Play ARM 64 v8a, android-33 | Google Play Intel x86_64 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9619390
    Xcode: 14.3/14E222b - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.18 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: Not Found
    react-native-macos: Not Found
    react-native-tvos:  0.71.8-0rc0 
  npmGlobalPackages:
    *react-native*: Not Found

### Issue and Reproduction Steps

When trying to build a project for testing with Detox there are multiple issues with merging native libs.
The command is executed from the root.
`cd android && ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release`
The same happens for debug test builds too. 
But commands `./gradlew assembleRelease` and `./gradlew assembleDebug` work

```
Execution failed for task ':lottie-react-native:mergeDebugAndroidTestNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeNativeLibsTask$MergeNativeLibsTaskWorkAction
   > 2 files found with path 'lib/arm64-v8a/libc++_shared.so' from inputs:
      - /Users/tikhonov/.gradle/caches/transforms-3/0b37af82917407f73b03a0e2024b8b1b/transformed/jetified-react-android-0.71.8-0rc0-debug/jni/arm64-v8a/libc++_shared.so
      - /Users/tikhonov/.gradle/caches/transforms-3/45571d5b399951265359e1bf8c7ee8f8/transformed/jetified-fbjni-0.3.0/jni/arm64-v8a/libc++_shared.so
```
The same happens for other libraries 
```
Execution failed for task ':react-native-community_netinfo:mergeDebugAndroidTestNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeNativeLibsTask$MergeNativeLibsTaskWorkAction
   > 2 files found with path 'lib/arm64-v8a/libc++_shared.so' from inputs:
      - /Users/tikhonov/.gradle/caches/transforms-3/0b37af82917407f73b03a0e2024b8b1b/transformed/jetified-react-android-0.71.8-0rc0-debug/jni/arm64-v8a/libc++_shared.so
      - /Users/tikhonov/.gradle/caches/transforms-3/45571d5b399951265359e1bf8c7ee8f8/transformed/jetified-fbjni-0.3.0/jni/arm64-v8a/libc++_shared.so
```

**Comment by douglowder** [comment]
@tikhonNikita do you have a small project with Android Detox tests that I can use to test fixes for this issue?

**Comment by tikhonNikita** [comment]
@douglowder please check this repo
https://github.com/tikhonNikita/tikhonNikita-detox-rn-tv-os
I've created a sample project that contains the latest react-native-tvos version, [react-native-netinfo](https://github.com/react-native-netinfo/react-native-netinfo) library that makes detox build fail.

**Comment by douglowder** [comment]
Thanks @tikhonNikita I am able to reproduce the issue and will investigate.

**Comment by douglowder** [comment]
This requires a patch to netinfo. Please try https://github.com/tikhonNikita/detox-rn-tv-os/pull/1 and see if that works

**Comment by tikhonNikita** [comment]
Thanks @douglowder! The same happens for lottie-react-native and some other libraries. Do you think I should use the same approach for them?

**Comment by tikhonNikita** [comment]
Thank you! As this issue occurs with many libraries. Here are some examples of them.
- @react-native-community/voice
- react-native-fast-image
- react-native-device-info

**Comment by douglowder** [comment]
Related issue in RN core: https://github.com/facebook/react-native/issues/34259

**Comment by douglowder** [comment]
@tikhonNikita I just verified that this is fixed in 0.71.8-0rc4.
